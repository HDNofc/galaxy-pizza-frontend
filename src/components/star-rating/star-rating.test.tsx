import { render } from '@testing-library/react';
import StarRating from './star-rating';

describe('<StarRating>', () => {
  it('отображает корректные количество звезд в зависимости от пропса `starAmount`', () => {
    const { container, rerender } = render(<StarRating rating={5} />);
    expect(container.firstChild).toMatchSnapshot();

    rerender(<StarRating starAmount={1} rating={5} />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('отображает корректно в зависимости от пропса `rating`', () => {
    const { container, rerender } = render(<StarRating rating={5} />);
    expect(container.firstChild).toMatchSnapshot();

    rerender(<StarRating rating={3.8} />);
    expect(container.firstChild).toMatchSnapshot();

    rerender(<StarRating rating={3.3} />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
