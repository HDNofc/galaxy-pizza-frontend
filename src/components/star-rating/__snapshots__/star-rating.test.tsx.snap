// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<StarRating> отображает корректно в зависимости от пропса \`rating\` 1`] = `
<div
  class="StarRating-sc-i2uazh-0 kKrkS"
>
  <div
    class="Star-sc-16jz84-0 gjzZhz"
  >
    <svg
      style="height: 20px;"
    >
      full.svg
    </svg>
  </div>
  <div
    class="Star-sc-16jz84-0 gjzZhz"
  >
    <svg
      style="height: 20px;"
    >
      full.svg
    </svg>
  </div>
  <div
    class="Star-sc-16jz84-0 gjzZhz"
  >
    <svg
      style="height: 20px;"
    >
      full.svg
    </svg>
  </div>
  <div
    class="Star-sc-16jz84-0 gjzZhz"
  >
    <svg
      style="height: 20px;"
    >
      full.svg
    </svg>
  </div>
  <div
    class="Star-sc-16jz84-0 gjzZhz"
  >
    <svg
      style="height: 20px;"
    >
      full.svg
    </svg>
  </div>
</div>
`;

exports[`<StarRating> отображает корректно в зависимости от пропса \`rating\` 2`] = `
<div
  class="StarRating-sc-i2uazh-0 kKrkS"
>
  <div
    class="Star-sc-16jz84-0 gjzZhz"
  >
    <svg
      style="height: 20px;"
    >
      full.svg
    </svg>
  </div>
  <div
    class="Star-sc-16jz84-0 gjzZhz"
  >
    <svg
      style="height: 20px;"
    >
      full.svg
    </svg>
  </div>
  <div
    class="Star-sc-16jz84-0 gjzZhz"
  >
    <svg
      style="height: 20px;"
    >
      full.svg
    </svg>
  </div>
  <div
    class="Star-sc-16jz84-0 gjzZhz"
  >
    <svg
      style="height: 20px;"
    >
      full.svg
    </svg>
  </div>
  <div
    class="Star-sc-16jz84-0 gjzZhz"
  >
    <svg
      style="height: 20px;"
    >
      zero.svg
    </svg>
  </div>
</div>
`;

exports[`<StarRating> отображает корректно в зависимости от пропса \`rating\` 3`] = `
<div
  class="StarRating-sc-i2uazh-0 kKrkS"
>
  <div
    class="Star-sc-16jz84-0 gjzZhz"
  >
    <svg
      style="height: 20px;"
    >
      full.svg
    </svg>
  </div>
  <div
    class="Star-sc-16jz84-0 gjzZhz"
  >
    <svg
      style="height: 20px;"
    >
      full.svg
    </svg>
  </div>
  <div
    class="Star-sc-16jz84-0 gjzZhz"
  >
    <svg
      style="height: 20px;"
    >
      full.svg
    </svg>
  </div>
  <div
    class="Star-sc-16jz84-0 gjzZhz"
  >
    <svg
      style="height: 20px;"
    >
      half.svg
    </svg>
  </div>
  <div
    class="Star-sc-16jz84-0 gjzZhz"
  >
    <svg
      style="height: 20px;"
    >
      zero.svg
    </svg>
  </div>
</div>
`;

exports[`<StarRating> отображает корректные количество звезд в зависимости от пропса \`starAmount\` 1`] = `
<div
  class="StarRating-sc-i2uazh-0 kKrkS"
>
  <div
    class="Star-sc-16jz84-0 gjzZhz"
  >
    <svg
      style="height: 20px;"
    >
      full.svg
    </svg>
  </div>
  <div
    class="Star-sc-16jz84-0 gjzZhz"
  >
    <svg
      style="height: 20px;"
    >
      full.svg
    </svg>
  </div>
  <div
    class="Star-sc-16jz84-0 gjzZhz"
  >
    <svg
      style="height: 20px;"
    >
      full.svg
    </svg>
  </div>
  <div
    class="Star-sc-16jz84-0 gjzZhz"
  >
    <svg
      style="height: 20px;"
    >
      full.svg
    </svg>
  </div>
  <div
    class="Star-sc-16jz84-0 gjzZhz"
  >
    <svg
      style="height: 20px;"
    >
      full.svg
    </svg>
  </div>
</div>
`;

exports[`<StarRating> отображает корректные количество звезд в зависимости от пропса \`starAmount\` 2`] = `
<div
  class="StarRating-sc-i2uazh-0 kKrkS"
>
  <div
    class="Star-sc-16jz84-0 gjzZhz"
  >
    <svg
      style="height: 20px;"
    >
      full.svg
    </svg>
  </div>
</div>
`;
